### ISSUE TITLE
Memory enhancement - cache versus actual usage

### ISSUE THREAD

**Issue by OBerghmans** [timdaman/check_docker#26]
First of all, nice script! It helps me alot. I've been monitoring several of my docker containers using your script.

For one container I noticed very high memory usage using the script (>5gb), while 'docker stats' only reported 250mb. After some troubleshooting I noticed the use of memory_stats>usage (line 283) in your script. This corresponds with the file /sys/fs/cgroup/memory/docker/container_id/memory.usage_in_bytes. This memory usage statistic seems to include the total cache as well. 

When I looked up the value of the cache and subtracted it from the above memory usage, I got the 250mb I was seeing in the 'docker stats'.

In summary, I changed line 283 to:
`usage = (inspection['memory_stats']['usage'] - inspection['memory_stats']['stats']['total_cache']) / UNIT_ADJUSTMENTS[units]`

I realize that in some cases you also want to monitor the cache usage, or at least have it included in the memory stats. Perhaps an enhancement could be to add an extra parameter that tells the memory check to include or exclude the cache?

**Comment by timdaman** [comment]
Thank you, I am happy to hear it has come in handy for you.

Thanks for great observation! I totally missed that. I am not a Go programmer but I took a peek at what the Docker engine was doing and it looks to me to match what you found. 

https://github.com/docker/docker-ce/blob/e947e4d4f1a55589a3eb4f049f51ddeddaf8c2da/components/cli/cli/command/container/stats_helpers.go

I have never come across a need to monitor a processes/containers cache usage, that is more a system wide stat. Frankly it would not make much sense to associate with a process since it is shared between processes/containers so it would be hard to deduce where the usages should be assigned.

Would you like to submit a PR for this improvement? I am fine with doing it myself from your notes but I wanted to give you the opportunity to take credit via the merge. As part of the change the unit tests will need to be expanded but that should be fairly simple.

**Comment by OBerghmans** [comment]
I will take a look soon, but no promises since I'm in a busy period.

**Comment by timdaman** [comment]
I am happy to patch it. I just wanted to give you the first crack at it.

**Comment by timdaman** [comment]
I think I have something that will work for you. Check out my 'prep_for_v2' branch.
https://github.com/timdaman/check_docker/tree/prep_for_v2
I still have a bit of work left to do on this branch but I would really appreciate your feedback before I merge to master.

**Comment by OBerghmans** [comment]
👍 Looks good! In terms of the adjustment in memory, it is similarly to what I did. Even better, because I forgot to adjust the % part.

**Comment by timdaman** [comment]
FYI, I just merged v2.0.0 into master. I am going to wait a day before pushing to pypi to give bugs a chance to bubble up.
Thanks for pointing out the mistake.

**Comment by OBerghmans** [comment]
Great! Thanks for the update. I updated it with the new version on my side and all looks OK (after changing memory unit from m to MB ofcourrse).
And again, thanks for the great script. I'm happy I could help.
